File: 704-binary-search/704. Binary Search.java
class Solution {
    public int search(int[] nums, int target) {
        
        int n = nums.length-1;
        int i = 0, j = n;

        while(i <= j)
        {
            int mid = i + (j-i)/2;
            if(nums[mid] == target)return mid;

            if(nums[mid] < target)i = mid+1;
            else j = mid-1;
        }

        return -1;
    }
}

File: 704-binary-search/704. Binary Search.cpp
class Solution {
public:
    int search(vector<int>& vec, int target) {
        sort(vec.begin(), vec.end());
        int n = vec.size();
        int low = 0, high = n-1;

        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (vec[mid] == target)return mid;

            if (vec[mid] > target)high = mid-1;

            else low = mid+1;
        }

        return -1;
    }
};